sqllogictest/src/runner.rs

```diff
@@ -875,6 +877,35 @@ impl<D: AsyncDB, M: MakeConnection<Conn = D>> Runner<D, M> {
         }
     }
 
+    /// Run a single record with retry.
+    pub async fn run_async_with_retry(
+        &mut self,
+        record: Record<D::ColumnType>,
+    ) -> Result<RecordOutput<D::ColumnType>, TestError> {
+        let retry = match &record {
+            Record::Statement { retry, .. } => retry.clone(),
+            Record::Query { retry, .. } => retry.clone(),
+            _ => None,
+        };
+
+        if retry.is_none() {
+            return self.run_async(record).await;
+        }
+        let retry = retry.unwrap();
+        let mut last_error = None;
+        for _ in 0..retry.attempts {
+            let result = self.run_async(record.clone()).await;
+            if result.is_ok() {
+                return result;
+            }
```

Comment:
Do we distinguish connection error and application error? I think it's not worth retrying if there's connection error.

Reply:
Considering that tests with `retry` should only take a very small proportion, I don't think it's necessary.